### PR TITLE
feat: adds okta verify number challenge

### PR DESCRIPTION
**Pull request by magizh-okta**
* If api responds with number challenge during okta verify challenge, UI displays number challenge.

* Updates lastAuthTransaction during polling. (needs authn changes)

https://okta.app.box.com/s/yhybly12s4jne89fsoc41rgkqxpmd2g8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "@commitlint/cli": "^8.1.0",
     "@commitlint/config-conventional": "^8.1.0",
     "@commitlint/travis-cli": "^8.1.0",
-    "@okta/okta-auth-js": "~2.6.3",
+    "@okta/okta-auth-js": "2.9.0-alpha.g0a3ff20",
```
Comment (mauriciocastillosilva-okta) [inline package.json]: reminder to update to a release version

File: packages/@okta/i18n/dist/properties/login.properties [hunk 2]
```diff
@@ -484,6 +484,8 @@ oktaverify.sent = Push sent!
 oktaverify.rejected = You have chosen to reject this login.
 oktaverify.timeout = Your push notification has expired.
 oktaverify.warning = Haven't received a push notification yet? Try opening the Okta Verify App on your phone.
+oktaverify.numberchallenge.instruction = On your phone tap, <span class="challenge-number">{0}</span> on Okta Verify prompt to continue.
```
Comment (mauriciocastillosilva-okta) [inline packages/@okta/i18n/dist/properties/login.properties]: there seems to be an extra comma in this sentence. It should be `On your phone tap ## on Okta Verify...`, or if you need a comma, it should be after phone `On your phone, tap ## on Okta Verify...`

File: src/MfaVerifyController.js [hunk 3]
```diff
@@ -123,6 +124,20 @@ function (Okta, BaseLoginController, TOTPForm, YubikeyForm, SecurityQuestionForm
         }
       }
 
+      this.listenTo(this.options.appState, 'change:isWaitingForNumberChallenge',
+        function (state, isWaitingForNumberChallenge) {
+          if (isWaitingForNumberChallenge || this.options.appState.get('lastAuthResponse').status === 'SUCCESS') {
```
Comment (mauriciocastillosilva-okta) [inline src/MfaVerifyController.js]: a couple of questions on this line
1. Is this called everytime we are setting `lastAuthResponse`? or only when the value actually changes from false to true? Reason I am asking is that derived properties are calculated every time, so I want to double check when this is called (could be everytime since the property is recalculated)
2. I see that this also checks for SUCCESS, does that mean that, without number challenge, when we approve a push notification, we will hide the elements? If so, is that needed?

Comment (magizh-okta) [reply src/MfaVerifyController.js]: 1) No. The function is executed only when the value of isWaitingForNumberChallenge changes. ie, if polling doesn't return number challenge the value of isWaitingForNumberChallenge remains false and the function never gets executed. 

2)  The success check will kick in only in number challenge flow and we wont hide other elements in non number challenge flow in success state. 

this is how isWaitingForNumberChallenge  changes in different state 
a) primary auth - false 
b) mfa required - false 
c) mfa challenge  - false 
d) mfa challenge with number challenge data - true
e) success - false ( this is the state that I handle here)

File: src/models/Factor.js [hunk 4]
```diff
@@ -325,6 +328,7 @@ function (Okta, Q, factorUtil, Util, Errors, BaseLoginModel) {
                   };
                 }
                 return trans.poll(options).then(function (trans) {
+                  self.options.appState.set('lastAuthResponse', trans.data);
```
Comment (gowthamidommety-okta) [inline src/models/Factor.js]: Why was this added?

File: src/MfaVerifyController.js [hunk 5]
```diff
@@ -22,10 +22,11 @@ define([
   'views/mfa-verify/PushForm',
   'views/mfa-verify/PasswordForm',
   'views/mfa-verify/InlineTOTPForm',
+  'views/mfa-verify/NumberChallengeView',
```
Comment (gowthamidommety-okta) [inline src/MfaVerifyController.js]: Do we need this in this file?

File: src/views/mfa-verify/PushForm.js [hunk 6]
```diff
@@ -110,7 +130,9 @@ define(['okta', 'util/CookieUtil', 'util/Util'], function (Okta, CookieUtil, Uti
         });
         this.trigger('save', this.model);
         warningTimeout = Util.callAfterTimeout(_.bind(function () {
-          this.showWarning(Okta.loc('oktaverify.warning', 'login'));
+          if (!this.options.appState.get('isWaitingForNumberChallenge')) {
```
Comment (gowthamidommety-okta) [inline src/views/mfa-verify/PushForm.js]: A UX question - do we need a similar warning for number challenge?

File: src/MfaVerifyController.js (hunk 5 above)
Comment (magizh-okta) [reply src/MfaVerifyController.js]: actually no will remove it. my bad!

File: src/models/Factor.js (hunk 4 above)
Comment (magizh-okta) [reply src/models/Factor.js]: This is after the last poll -> success. The callback doesn't get  called in this case.

File: src/MfaVerifyController.js (hunk 3 above)
Comment (mauriciocastillosilva-okta) [reply src/MfaVerifyController.js]: 1. thanks
2. Got it. So this is a success that will follow showing the number challenge, not any other flow. Makes sense
👍

File: src/views/mfa-verify/PushForm.js (hunk 6 above)
Comment (magizh-okta) [reply src/views/mfa-verify/PushForm.js]: discussed with Steven and decided to hold of on any custom warning messages for number challenge as current text in the UI is descriptive enough.

File: src/models/Factor.js (hunk 4 above)
Comment (gowthamidommety-okta) [reply src/models/Factor.js]: Is this needed to hide the number challenge?

**Comment by magizh-okta** [comment]
Taken care in path version https://github.com/okta/okta-signin-widget/pull/884/
